### PR TITLE
Upgrade Docker SDK, make necessary code changes to support new version.

### PR DESCRIPTION
**Pull request by johngriebel**
Issue #364 - Upgrade Python Docker SDK
Signed-off-by: John Griebel <jgriebel@3blades.io>

### CODE REVIEW

File: infrastructure/models.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 from django.conf import settings
 from django.db import models
 from django.urls import reverse
-from docker import Client
+import docker
+# from docker import Client
```
Comment (jgwerner) [inline infrastructure/models.py]: @jkp85 @johngriebel Maybe a stickler but we shouldn't leave commented code in master. What do you think?

Comment (johngriebel) [reply infrastructure/models.py]: Whoops, missed that. I'll take it out. Nice catch

**Comment by johngriebel** [comment]
@jgwerner Okay sounds good. Let me know when it's been validated and we can merge.

**Comment by jgwerner** [comment]
@johngriebel I can't spawn a server from the UI. This is the same behavior I encountered.

launch_server_error

Celery instance throws the exception below. It does look like it instantiates the `client` object correctly, but for some reason when using docker.client.from_env(), such as was the case with the POC, doesn't work with app-backend:

```
[2017-09-12 17:55:55,481: INFO/ForkPoolWorker-14] Unexpected error trying to terminate a server
[2017-09-12 17:55:55,484: ERROR/ForkPoolWorker-14] Task servers.tasks.terminate_server[620d698a-4dca-43c1-b278-3cc9f9102998] raised unexpected: AttributeError("'DockerClient' object has no attribute 'remove_container' In Docker SDK for Python 2.0, this method is now on the object APIClient. See the low-level API section of the documentation for more details.",)
Traceback (most recent call last):
  File "/srv/env/lib/python3.6/site-packages/celery/app/trace.py", line 374, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/srv/env/lib/python3.6/site-packages/celery/app/trace.py", line 629, in __protected_call__
    return self.run(*args, **kwargs)
  File "/srv/app/servers/tasks.py", line 24, in terminate_server
    server_action('terminate', server)
  File "/srv/app/servers/tasks.py", line 9, in server_action
    getattr(spawner, action)()
  File "/srv/app/servers/spawners.py", line 241, in terminate
    self.client.remove_container(self.server.container_name)
  File "/srv/env/lib/python3.6/site-packages/docker/client.py", line 187, in __getattr__
    raise AttributeError(' '.join(s))
AttributeError: 'DockerClient' object has no attribute 'remove_container' In Docker SDK for Python 2.0, this method is now on the object APIClient. See the low-level API section of the documentation for more details.
```

**Pull request merged**